### PR TITLE
Add VC++ Debug redist deployment to `deploy-ebpf.ps1`.

### LINKED ISSUE #2533: deploy-ebpf.ps1 doesn't deploy redist
### Describe the bug

After using the deploy method at 
https://github.com/microsoft/ebpf-for-windows/blob/main/docs/InstallEbpf.md#method-2-install-files-you-built-yourself

Running unit_tests.exe fails missing MSVCP140D_ATOMIC_WAIT.dll

### OS information

_No response_

### Steps taken to reproduce bug

Follow the instructions at 
https://github.com/microsoft/ebpf-for-windows/blob/main/docs/InstallEbpf.md#method-2-install-files-you-built-yourself

Run unit_tests.exe in the VM.

### Expected behavior

unit_tests.exe passes

### Actual outcome

unit_tests.exe fails

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by gtrevi**
Closes #2533

## Description

This PR:

- Fixes incorrect `--local` parameter parsing.
- Adds the missing step for deploying the VC++ Debug redist to `deploy.ps1`.

## Testing

Local, CI/CD

## Documentation

Updated `InstallEbpf.md`.

### CODE REVIEW

File: scripts/deploy-ebpf.ps1.in [hunk 1]
```diff
@@ -283,7 +284,7 @@ OPTIONS:
 }
 
 $build_directory="$PSScriptRoot"
-if ($vm -eq $null) {
+if ($null -eq $vm) {
```
Comment (dthaler) [inline scripts/deploy-ebpf.ps1.in:287]: Why this change?  Seems unnecessary for this PR

File: scripts/deploy-ebpf.ps1.in [hunk 2]
```diff
@@ -297,6 +298,21 @@ if ($vm -eq $null) {
       Copy-Item "$source_path" -Destination "$destination_path" -ErrorAction Stop
    }
 
+   if ($copy_redist) {
+    Write-Host "Copying the VC++ Redist (Debug) files from `"$vc_redist_dir`" to `"$destination_directory`""
```
Comment (dthaler) [inline scripts/deploy-ebpf.ps1.in:302]: indentation of these lines doesn't match rest of file

File: scripts/deploy-ebpf.ps1.in (hunk 1 above)
Comment (gtrevi) [reply scripts/deploy-ebpf.ps1.in:287]: Best practice in PS scipts.

Comment (dthaler) [reply scripts/deploy-ebpf.ps1.in:287]: Please provide a reference for that claim, since I don't think it's helpful.

Comment (gtrevi) [reply scripts/deploy-ebpf.ps1.in:287]: agree as discussed, changing this also in other scripts.

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]
